godot-kotlin/godot-library/src/nativeCore/kotlin/godot/core/type/Color.kt

```diff
@@ -0,0 +1,806 @@
+    }
+
+
+    //INTEROP
+    override fun getRawMemory(memScope: MemScope): COpaquePointer {
+        val value = cValue<godot_color_layout> {
+            r = this@Color.r.toFloat()
+            g = this@Color.g.toFloat()
+            b = this@Color.b.toFloat()
+            a = this@Color.a.toFloat()
+        }
+        return value.getPointer(memScope)
+    }
+
+    override fun setRawMemory(mem: COpaquePointer) {
+        val value = mem.reinterpret<godot_color_layout>().pointed
+        r = value.r.toRealT()
+        g = value.g.toRealT()
+        b = value.b.toRealT()
+        a = value.a.toRealT()
+    }
+
+    //API
+    /**
+     * Returns a new color resulting from blending this color over another.
+     * If the color is opaque, the result is also opaque. The second color may have a range of alpha values.
+     */
+    fun blend(over: Color): Color {
+        val res = Color()
+        val sa = 1.0 - over.a
+        if (res.a == 0.0) {
+            return Color(0.0, 0.0, 0.0, 0.0)
+        } else {
+            res.r = (r * a * sa + over.r * over.a) / res.a
+            res.g = (g * a * sa + over.g * over.a) / res.a
+            res.b = (b * a * sa + over.b * over.a) / res.a
+        }
+        return res
+    }
+
+    /**
+     * Returns the most contrasting color.
+     */
+    fun contrasted(): Color {
+        val c = this
+        c.contrast()
+        return c
+    }
+
+
+    internal fun contrast() {
+        r = (r + 0.5).rem(1.0)
+        g = (g + 0.5).rem(1.0)
+        b = (b + 0.5).rem(1.0)
+    }
+
+    /**
+     * Returns a new color resulting from making this color darker by the specified percentage (ratio from 0 to 1).
+     */
+    fun darkened(amount: RealT): Color {
+        return Color(
+            r * (1.0 - amount),
+            g * (1.0 - amount),
+            b * (1.0 - amount),
+            a
+        )
+    }
+
+    /**
+     * Constructs a color from an HSV profile. h, s, and v are values between 0 and 1.
+     */
+    fun fromHsv(h: RealT, s: RealT, v: RealT, a: RealT): Color {
+        var h2 = (h * 360.0).rem(360.0)
+
+        if (h2 < 0) h2 += 360
+
+        val finalH = h2 / 60
+        val c = v * s
+        val x = c * (1.0 - (finalH.rem(2) - 1))
+        val rgbTriple = when (finalH.toInt()) {
+            0 -> Triple(c, x, 0.0)
+            1 -> Triple(x, c, 0.0)
+            2 -> Triple(0.0, c, x)
+            3 -> Triple(0.0, x, c)
+            4 -> Triple(x, 0.0, c)
+            5 -> Triple(c, 0.0, x)
+            else -> Triple(0.0, 0.0, 0.0)
+        }
+
+        val m = v - c
+        return Color(m + rgbTriple.first, m + rgbTriple.second, m + rgbTriple.third, a)
+    }
+
+    /**
+     * Returns the color’s grayscale representation.
+     * The gray value is calculated as (r + g + b) / 3.
+     */
+    fun gray(): RealT = (r + g + b) / 3.0
+
+    /**
+     * Returns the inverted color (1 - r, 1 - g, 1 - b, a).
+     */
+    fun inverted(): Color {
+        val c = this
```

Comment:
Same here.

Reply:
this modifies the current object, I think we should return a new copy.